zadanie

BartoszKlocek opened the pull request.

BartoszKlocek committed:
wymaganie2
Zabezpiecz metodę przed nullami. Gdy w parametrze zostanie przekazany null, wtedy metoda powinna zwrócić „Witaj, mój przyjacielu”

BartoszKlocek committed:
wymaganie3
Gdy imie zostanie wpisane wielkimi literami np.”ADAM”, wtedy metoda powinna zwrócić „WITAJ ADAM!”

BernardBoguszewski commented on src/main/java/com/example/demo/Greetings.java:
```diff
@@ -4,8 +4,14 @@
+        }
+
+
+        else if (name.equals(name.toUpperCase())){
```
formatowanie kodu - ctrl + alt + L

BartoszKlocek committed:
wymaganie4
Gdy zostaną podane dwa imiona po przecinku np. „Adam,Iza”, metoda ma zwrócić „Adam i Iza, witajcie!”.

BartoszKlocek committed:
wymaganie5
Gdy zostaną podane trzy imiona np. „Adam,Iza,Robert”, metoda ma zwrócić „Adam, Iza i Robert, witajcie!”

BartoszKlocek committed:
back fix - wymaganie4 i wymaganie5

BernardBoguszewski commented on src/main/java/com/example/demo/Greetings.java:
```diff
@@ -18,10 +18,15 @@ else if (name.contains(",")) {
             if (name.split(",").length == 2) {
 
                 String[] tab = name.split(",");
+                tab[0]=tab[0].trim();
```
warto wydzielić trimowanie do osobnej metody pomocniczej

BartoszKlocek committed:
wymaganie 6
Gdy jedno z trzech imion będzie napisane wielkimi literami, np.”Adam, Iza, ROBERT”, wtedy metoda ma zwrócić „Adam i Iza, witajcie! WITAJ ROBERT!”.

BernardBoguszewski commented on src/main/java/com/example/demo/Greetings.java:
```diff
@@ -7,35 +7,66 @@ public String greet(String name) {
+    }
+
+    public static String greetingsOneUpperCase(String[] tab) {
+        if (tab[0].equals(tab[0].toUpperCase())
```
bardziej rozbudowane warunki możesz wyciągnąć do metod pomocniczych

BartoszKlocek committed:
wymaganie 7
Jeżeli w zmiennej name znajdują się cyfry, metoda ma zwrócić „Z liczbami się nie witam.”.
